### PR TITLE
Issue #658 TransformFactory and related tests

### PR DESCRIPTION
**Pull request by lovemaths**

### CODE REVIEW

File: src/Microsoft.IdentityModel.Xml/TransformFactory.cs [hunk 1]
```diff
@@ -52,12 +52,41 @@ public static TransformFactory Default
...
+
+        /// <summary>
         /// Gets a XML transform that modifies a XmlTokenStream.
         /// </summary>
         /// <param name="transform">the name of the transform.</param>
         /// <returns><see cref="Transform"/></returns>
         public virtual Transform GetTransform(string transform)
         {
+            if (!IsSupportedTransform(transform))
+                throw LogExceptionMessage(new NotSupportedException($"transform not supported: '{transform}'."));
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Xml/TransformFactory.cs]: Strings should be in LogMessages.cs

File: src/Microsoft.IdentityModel.Xml/TransformFactory.cs [hunk 2]
```diff
@@ -71,6 +100,9 @@ public virtual Transform GetTransform(string transform)
         /// <returns><see cref="CanonicalizingTransfrom"/></returns>
         public virtual CanonicalizingTransfrom GetCanonicalizingTransform(string transform)
         {
+            if (!IsSupportedCanonicalizingTransfrom(transform))
+                throw LogExceptionMessage(new NotSupportedException($"transform not supported: '{transform}'."));
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Xml/TransformFactory.cs]: Strings in LogMessages.cs

File: test/Microsoft.IdentityModel.Xml.Tests/TransformFactoryTest.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    public class TransformFactoryTest
+    {
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+        [Theory, MemberData("GetTransformTestTheoryData")]
+#pragma warning restore CS3016 // Arrays as attribute arguments is not CLS-compliant
+        public void GetTransformTest(TransformTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.GetTransformTest", theoryData);
+
+            var transformFactory = new TransformFactory();
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Xml.Tests/TransformFactoryTest.cs]: Have a look at the code below, we get better error messaging if we add CompareContext to ProcessNoException and ProcessException.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/test/Microsoft.IdentityModel.Xml.Tests/ReferenceTests.cs#L70

**Comment by lovemaths** [comment]
@brentschmaltz Made the changes.

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/ClaimProperties.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+//------------------------------------------------------------------------------
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/ClaimProperties.cs]: It looks like you picked up on changes to PR: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/659

Can you scope this PR to just your changes.

**Comment by lovemaths** [comment]
@brentschmaltz Rebased against dev

**Comment by brentschmaltz** [comment]
:shipit: as an intermediate step.

**Pull request merged**